AutomatedTesting/Gem/PythonTests/Atom/tests/rhi_validation/RHIValidation_Vulkan_DefaultLevel.py

```diff
@@ -0,0 +1,52 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+class Checks:
+    enter_game_mode                = ("Entered game mode",              "Failed to enter game mode")
+    no_rhi_validation_errors_found = ("No RHI validation errors found", "Found RHI validation errors")
+    exit_game_mode                 = ("Exited game mode",               "Failed to exit game mode")
+
+def RHIValidation_Vulkan_DefaultLevel():
+    
+    import azlmbr.legacy.general as general
+    
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+    from editor_python_test_tools.utils import Tracer
+    
+    from vulkan_skip_errors import VulkanValidationErrors
+    
+    # Constants
+    FRAMES_IN_GAME_MODE = 200
+
+    # 1) Start the Tracer to catch any errors and warnings
+    with Tracer() as section_tracer:
+        helper.init_idle()
+
+        # 2) Load the level
+        helper.open_level("", "DefaultLevel")
+
+        # 3) Enter game mode
+        helper.enter_game_mode(Checks.enter_game_mode)
+        
+        # 4) Wait in game mode some frames to let cloth simulation run
+        general.idle_wait_frames(FRAMES_IN_GAME_MODE)
+        
+        # 5) Exit game mode
+        helper.exit_game_mode(Checks.exit_game_mode)
+
+    # 5) Verify there are no rhi validation errors in the logs
+    has_rhi_validation_errors = False
+    for error_msg in section_tracer.errors:
+        if VulkanValidationErrors.CheckError(error_msg.window, f"{error_msg}"):
+            has_rhi_validation_errors = True
+            Report.info(f"RHI Validation error found: {error_msg}")
+    Report.result(Checks.no_rhi_validation_errors_found, not has_rhi_validation_errors)
```

Comment:
@Kadino is this the correct way to capture vulkan validation errors?

I was wondering if it might be too late to start listening to the errors, since the test is executed after the editor has been opened, I was wondering if there is some error during initialization of the RHI if we are not catching it. If not, what alternative would you suggest?

Cheers

Reply:
The test will be started by python after loading the editor, which I expect includes (or possibly races with) the initialization of Vulkan/RHI/Atom. If the validation issues are occurring before the call to open level, then it will not reliably catch them. If it was possible to delay emitting errors until after the test started, that could assure the test's tracer encounters them. This sounds like a lot of functional work to chase this down, so perhaps unapproachable.

Perhaps the call to CheckError could always be called in the test, to see if the error window is populated? (in a look-before-you-leap safe way that won't explode when the error window isn't present) This would likely require using some sort of unique string or tag to match in the error window that identifies it as a vulkan error, as that window can contain messages from other systems.

...this would rely more heavily on the error window being correctly populated, yet it would also prevent an existing issue where the error window gets populated with a Vulkan error that accidentally didn't get sent to the log. Frankly looking both at the log and the error window seems appropriate, and not gating error reporting on only the log contents which may have been missed.

Reply:
Thanks for the reply @Kadino. Yeah, it's what I thought, it will pick up vulkan errors before calling open level, which is very valuable in itself, many errors happen when it actually starts rendering something, it's just we're missing the rhi initialization ones.

So I'm going to proceed to submit this, as it's a good start point, and I'll create another task to see how we can catch the initialization ones. I'll reach you at some point to discuss possibilities.

Thanks!